greenmail-core/src/test/java/com/icegreen/greenmail/test/SendReceiveWithInternationalAddress.java

```diff
@@ -0,0 +1,167 @@
+package com.icegreen.greenmail.test;
+
+import com.icegreen.greenmail.junit.GreenMailRule;
+import com.icegreen.greenmail.util.GreenMailUtil;
+import com.icegreen.greenmail.util.ServerSetupTest;
+import com.sun.mail.util.PropUtil;
+import org.junit.Rule;
+import org.junit.Test;
+
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeUtility;
+
+import java.io.UnsupportedEncodingException;
+import java.util.Date;
+import java.util.Properties;
+
+import static org.junit.Assert.assertEquals;
+
+public class SendReceiveWithInternationalAddress {
+
+    static final Properties properties;
+
+    static {
+
+        properties = new Properties();
+        properties.put("mail.mime.address.strict", Boolean.FALSE.toString());
+    }
+
+    @Rule
+    public final GreenMailRule greenMail = new GreenMailRule(ServerSetupTest.SMTP_IMAP);
+
+    @Test
+    public void testSend() throws MessagingException, UnsupportedEncodingException {
+
+        Session session = GreenMailUtil.getSession(ServerSetupTest.SMTP, properties);
+        MimeMessage mimeMessage = new InternationalizedMimeMessage(session);
+        mimeMessage.setSubject("subject");
+        mimeMessage.setSentDate(new Date());
+        mimeMessage.setFrom("múchätįldé@tìldę.oœ");
+        mimeMessage.setRecipients(Message.RecipientType.TO, "用户@例子.广告");
+        mimeMessage.setRecipients(Message.RecipientType.CC, "θσερεχα@μπλε.ψομ");
+        mimeMessage.setRecipients(Message.RecipientType.BCC, "राममो@हन.ईन्फो");
+
+        // The body text needs to be encoded if it contains non us-ascii characters
+        mimeMessage.setText(MimeUtility.encodeText("用户@例子"));
+
+        GreenMailUtil.sendMimeMessage(mimeMessage);
+
+        // Decoding the body text to verify equality
+        String decodedText = MimeUtility.decodeText(GreenMailUtil.getBody(greenMail.getReceivedMessages()[0]));
+        assertEquals("用户@例子", decodedText);
+    }
+
+    private static class InternationalizedMimeMessage extends MimeMessage {
+
+        public InternationalizedMimeMessage(Session session) {
+            super(session);
+        }
+
+        // Current Java Mail version does not respect the "mail.mime.address.strict"
+        // value when setting email addresses in a MimeMessage object (calls
+        // InternetAddress.parse(String), which will always validate the
+        // address based on rfc822) so we need to override this method to
+        // make this example work.
+        @Override
+        public void setRecipients(Message.RecipientType type, String addresses) throws MessagingException {
+
+            if (type == RecipientType.NEWSGROUPS) {
```

Comment:
And please add a comment to InternationalizedMimeMessage saying that this is a mock message that doesn't implement the full functionality from MimeMessage. Maybe rename it to MockInternationalizedMimeMessage